packages/packages_test.go

```diff
@@ -442,13 +443,234 @@ func TestPackagesFilter(t *testing.T) {
 
 	for _, c := range cases {
 		t.Run(c.Title, func(t *testing.T) {
-			result := c.Filter.Apply(context.Background(), packages)
+			result, err := c.Filter.Apply(context.Background(), packages)
+			require.NoError(t, err)
+			assertFilterPackagesResult(t, c.Expected, result)
+		})
+	}
+}
+
+func TestPackagesSpecMinMaxFilter(t *testing.T) {
+	filterTestPackages := []filterTestPackage{
```

Comment:
Could you add a test case with a patch version and check that it is included with spec.max set to its minor? For example a package with `FormatVersion: "2.2.1"`, and check that is included when `spec.max` is `2.2`.

Reply:
Good catch! Current implementation would not include a package with format_version 2.2.1
with the query spec.max 2.2

Reply:
Updated in https://github.com/elastic/package-registry/pull/1058